hibernate-core/src/test/java/org/hibernate/orm/test/mapping/identifier/composite/CompositeInheritanceWorkingTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.mapping.identifier.composite;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.MappedSuperclass;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.Jira;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * This test works for some reason...
+ */
+@DomainModel(
+		annotatedClasses = {
+				CompositeInheritanceWorkingTest.TupAbstractEntity.class,
+				CompositeInheritanceWorkingTest.DummyEntity.class,
+				CompositeInheritanceWorkingTest.FooEntity.class, // And here the class is called FooEntity and this works for some reason
+		}
+)
+@ServiceRegistry(
+		settings = {
+				// For your own convenience to see generated queries:
+				@Setting(name = AvailableSettings.SHOW_SQL, value = "true"),
+				@Setting(name = AvailableSettings.FORMAT_SQL, value = "true"),
+				// @Setting( name = AvailableSettings.GENERATE_STATISTICS, value = "true" ),
+		}
+)
+@SessionFactory
+@Jira("HHH-19076")
+public class CompositeInheritanceWorkingTest {
```

Comment:
No need for a "working" test case, if you really wish to include both versions of the mappings you can do so in a single test class and separate test methods.

Reply:
I've simply (re)used commit with both test cases, but you are right. I will simply remove test that was passing without any changes.